src/OpenLoco/src/Paint/PaintTrack.cpp

```diff
@@ -26,6 +26,97 @@ namespace OpenLoco::Paint
     static loco_global<uint8_t, 0x00113605E> _trackTunnel;
     static loco_global<uint8_t, 0x00522095> _byte_522095;
 
+    struct TrackPaintCommon
+    {
+        ImageId trackBaseImageId;         // 0x01135F26 with colours and image index set to base of trackObject image table
+        ImageId bridgeColoursBaseImageId; // 0x01135F36 with only the colours set (image index not set!)
+        uint8_t tunnelType;               // 0x0113605E
+    };
+
+    constexpr std::array<World::Pos3, 4> kStraightBoundingBoxOffsets = {
+        World::Pos3{ 2, 5, 0 },
+        World::Pos3{ 5, 2, 0 },
+        World::Pos3{ 2, 5, 0 },
+        World::Pos3{ 5, 2, 0 },
+    };
+    constexpr std::array<World::Pos3, 4> kStraightBoundingBoxSizes = {
+        World::Pos3{ 28, 22, 1 },
+        World::Pos3{ 22, 28, 1 },
+        World::Pos3{ 28, 22, 1 },
+        World::Pos3{ 22, 28, 1 },
+    };
+    constexpr std::array<std::array<uint32_t, 3>, 4> kStraightImageIndexOffsets = {
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastNE, TrackObj::ImageIds::Style0::straightSleeperNE, TrackObj::ImageIds::Style0::straightRailNE },
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastSW, TrackObj::ImageIds::Style0::straightSleeperSW, TrackObj::ImageIds::Style0::straightRailSW },
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastNE, TrackObj::ImageIds::Style0::straightSleeperNE, TrackObj::ImageIds::Style0::straightRailNE },
+        std::array<uint32_t, 3>{ TrackObj::ImageIds::Style0::straightBallastSW, TrackObj::ImageIds::Style0::straightSleeperSW, TrackObj::ImageIds::Style0::straightRailSW },
+    };
+    constexpr std::array<uint8_t, 4> kStraightBridgeEdges = {
+        0b0101,
+        0b1010,
+        0b0101,
+        0b1010,
+    };
+    constexpr uint8_t kStraightBridgeQuarters = 0xF;
+    constexpr std::array<uint8_t, 4> kStraightTunnelsEdges = {
+        0b0101,
+        0b1010,
+        0b0101,
+        0b1010,
+    };
+    constexpr std::array<SegmentFlags, 4> kStraightSegments = {
+        SegmentFlags::x0y1 | SegmentFlags::x1y1 | SegmentFlags::x2y1,
```

Comment:
Can't we just have definitions for a single view, and rotate them at run/compile time?

Reply:
Yes i was thinking compile time rotate for most of the fields